src/routes/assemblies/[id]/members/[groupby]/+page.svelte

```diff
@@ -1,8 +1,243 @@
 <script lang="ts">
+	import Header from '$components/Assemblies/Members/Header.svelte';
+	import Tab from '$components/Assemblies/Members/Tab.svelte';
+	import PoliticianProfile from '$components/PoliticianProfile/PoliticianProfile.svelte';
+	import {
+		Accordion,
+		AccordionItem,
+		Button,
+		Checkbox,
+		FormGroup,
+		Search
+	} from 'carbon-components-svelte';
+	import Minimize from 'carbon-icons-svelte/lib/Minimize.svelte';
+	import SearchLocate from 'carbon-icons-svelte/lib/SearchLocate.svelte';
+	import scrollama from 'scrollama';
+	import { onMount } from 'svelte';
+	import type { PoliticianGroupBy } from './+page.js';
+
 	export let data;
-	$: ({ groups } = data);
+	$: ({ groups, groupByTabs } = data);
+	$: currentPath = groupByTabs.find((e) => e.isActive)?.path ?? '';
+
+	let showFilter = true;
+	let searchQuery = '';
+	let isByDistrict = true;
+	let isByPartylist = true;
+
+	$: formattedSearchQuery = searchQuery.trim();
+	$: filteredGroup =
+		formattedSearchQuery === '' && isByDistrict && isByPartylist
+			? groups
+			: (groups.map((group) => {
+					if ('subgroups' in group) {
+						return {
+							...group,
+							subgroups: group.subgroups.map((subgroup) => {
+								return {
+									...subgroup,
+									members: subgroup.members.filter((member) => {
+										return (
+											(member.firstname + ' ' + member.lastname).includes(formattedSearchQuery) &&
+											((isByDistrict && member.candidateType === 'แบ่งเขต') ||
+												(isByPartylist && member.candidateType === 'บัญชีรายชื่อ'))
+										);
+									})
+								};
+							})
+						};
+					}
+					return group.name.includes(formattedSearchQuery);
+			  }) as PoliticianGroupBy);
+
+	let isMobile = false;
+	let mounted = false;
+	let currentCatg = '';
+	onMount(() => {
+		mounted = true;
+
+		showFilter = window.matchMedia(`(min-width: 672px)`).matches;
+		isMobile = !showFilter;
+		if (showFilter) {
+			currentCatg = (
+				groups[0].name +
+				'-' +
+				('subgroups' in groups[0] ? groups[0].subgroups[0].name : '')
+			).replace(/ /g, '-');
+
+			const scroller = scrollama();
+
+			scroller
+				.setup({
+					step: '.member-subcategory',
+					// @ts-expect-error Documentation บอกว่าใช้ string ที่มี px ได้ https://github.com/russellsamora/scrollama#:~:text=number%20(0%20%2D%201%2C%20or%20string%20with%20%22px%22)
+					offset: '128px'
+				})
+				.onStepEnter((response) => {
+					currentCatg = response.element.children[0].id;
+				});
+
+			return scroller.destroy;
+		}
+	});
 </script>
 
-<div class="whitespace-pre">
-	{JSON.stringify(groups, undefined, 2)}
+<Header {data} bind:searchQuery {mounted} />
+<Tab {data} />
+<div class="relative flex">
+	{#if showFilter}
+		<aside
+			class={[
+				'flex flex-col gap-4 px-6 bg-white',
+				'fixed left-0 right-0 top-0 bottom-0 z-40',
+				'md:sticky md:top-[80px] md:w-[250px] md:h-[calc(100dvh-80px)] md:flex-none md:z-0',
+				'member-aside'
+			].join(' ')}
```

Comment:
Is this for not having a too-long line? No need to change, I've just never seen this before lol.